app/controllers/post_controller.rb

```diff
@@ -1,9 +1,13 @@
 class PostsController < ApplicationController
   def index
-    @posts = Post.all
+    @user = User.find(params[:user_id])
+    @user_posts = Posts.where(author_id: @user.id)
+    @user_comments = Comment.where(user_id: @user.id)
+    @post_comments = Comment.where(post_id: @user_posts.ids)
   end
 
   def show
-    @post = Post.find(params[:id])
+    @post = Posts.find(params[:id])
+    @post_comments = Comment.where(post_id: @post)
   end
 end
```

Comment:
- [ ] Kindly take note of the change requested by the previous reviewer. I get an error page when I try to visit the post index or show page.

  ![image](https://user-images.githubusercontent.com/54848363/234668556-ed9c5ec6-d81f-4ce8-9dea-4d9dc3334fb8.png)

  #### How to fix this:
  - You have two files named `post_controller.rb` and `posts_controller.rb`. The `posts_controller.rb` file is mostly dormant. Kindly remove it.
  - Rename this file that contains your code from `post_controller.rb` to `posts_controller.rb`
  - You need to leave the previous code as it was during the previous review:

    ```
    class PostsController < ApplicationController
      def index
        @user = User.find(params[:user_id])
        @user_posts = Post.where(author_id: @user.id)
        @user_comments = Comment.where(user_id: @user.id)
        @post_comments = Comment.where(post_id: @user_posts.ids)
      end
    
      def show
        @post = Post.find(params[:id])
        @post_comments = @post.comments
      end
    end
    ```